bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleManager.java

```diff
@@ -0,0 +1,248 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    ProSyst Software GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.smarthome.automation.core;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.eclipse.smarthome.automation.Rule;
+import org.osgi.framework.BundleContext;
+
+/**
+ *
+ * @author Yordan Mihaylov - Initial Contribution
+ */
+public abstract class RuleManager {
+
+    public static final String ID_PREFIX = "rule_"; //$NON-NLS-1$
+    private static RuleEngine re;
+
+    private static int maxId;
+
+    public RuleManager(BundleContext bc) {
+        Set<RuleImpl> rules = loadRules();
+        if (rules == null) {
+            rules = new HashSet<RuleImpl>(20);
+        }
+        maxId = getMaxId(ID_PREFIX, rules) + 1;
+        if (re == null) {
+            re = new RuleEngine(bc);
+        }
+        for (Iterator<RuleImpl> it = rules.iterator(); it.hasNext();) {
+            re.setRule(it.next());
+        }
+    }
+
+    public void dispose() {
+        if (re != null) {
+            re.dispose();
+            re = null;
+        }
+    }
+
+    /**
+     *
+     * @see org.eclipse.smarthome.automation.RuleRegistry#getRule(java.lang.String)
+     */
+    public synchronized Rule getRule(String ruleUID) {
+        RuleImpl oldR = re.getRule(ruleUID);
+        if (oldR == null)
+            return null;
+        return new RuleImpl(oldR);
+    }
+
+    /**
+     * @see org.eclipse.smarthome.automation.RuleRegistry#getRules(java.lang.String)
```

Comment:
Hm, there isn't such a thing as ```org.eclipse.smarthome.automation.RuleRegistry#getRules(java.lang.String)```.
I'd suggest to remove this method (at least as long as we haven't defined any suitable filters yet).